gensim/sklearn_integration/sklearn_wrapper_gensim_w2vmodel.py

```diff
@@ -0,0 +1,116 @@
+        self.max_vocab_size = max_vocab_size
+        self.sample = sample
+        self.seed = seed
+        self.workers = workers
+        self.min_alpha = min_alpha
+        self.sg = sg
+        self.hs = hs
+        self.negative = negative
+        self.cbow_mean = int(cbow_mean)
+        self.hashfxn = hashfxn
+        self.iter = iter
+        self.null_word = null_word
+        self.trim_rule = trim_rule
+        self.sorted_vocab = sorted_vocab
+        self.batch_words = batch_words
+
+    def get_params(self, deep=True):
+        """
+        Returns all parameters as dictionary.
+        """
+        return {"size": self.size, "alpha": self.alpha, "window": self.window, "min_count": self.min_count,
+        "max_vocab_size": self.max_vocab_size, "sample": self.sample, "seed": self.seed,
+        "workers": self.workers, "min_alpha": self.min_alpha, "sg": self.sg, "hs": self.hs,
+        "negative": self.negative, "cbow_mean": self.cbow_mean, "hashfxn": self.hashfxn,
+        "iter": self.iter, "null_word": self.null_word, "trim_rule": self.trim_rule,
+        "sorted_vocab": self.sorted_vocab, "batch_words": self.batch_words}
+
+    def set_params(self, **parameters):
+        """
+        Set all parameters.
+        """
+        super(SklW2VModel, self).set_params(**parameters)
+        return self
+
+    def fit(self, X, y=None):
+        """
+        Fit the model according to the given training data.
+        Calls gensim.models.Word2Vec
+        """
+        self.gensim_model = models.Word2Vec(sentences=X, size=self.size, alpha=self.alpha,
+            window=self.window, min_count=self.min_count, max_vocab_size=self.max_vocab_size,
+            sample=self.sample, seed=self.seed, workers=self.workers, min_alpha=self.min_alpha,
+            sg=self.sg, hs=self.hs, negative=self.negative, cbow_mean=self.cbow_mean,
+            hashfxn=self.hashfxn, iter=self.iter, null_word=self.null_word, trim_rule=self.trim_rule,
+            sorted_vocab=self.sorted_vocab, batch_words=self.batch_words)
+        return self
+
+    def transform(self, words):
+        """
+        Return the word-vectors for the input list of words.
+        """
+        if self.gensim_model is None:
+            raise NotFittedError("This model has not been fitted yet. Call 'fit' with appropriate arguments before using this method.")
+
+        # The input as array of array
+        check = lambda x: [x] if isinstance(x, six.string_types) else x
+        words = check(words)
+        X = [[] for _ in range(0, len(words))]
+
+        for k, v in enumerate(words):
+            word_vec = self.gensim_model[v]
+            X[k] = word_vec
+
+        return np.reshape(np.array(X), (len(words), self.size))
+
+    def partial_fit(self, X):
+        """
+        Update model using newly added sentences.
+        """
+        if self.gensim_model is None:
+            self.gensim_model = models.Word2Vec(size=self.size, alpha=self.alpha,
```

Comment:
Since `train()` (below) requires additional model vocab initialization (as by `build_vocab()`), I can't see this path, where the model doesn't already exist, possibly working. More generally, I'm not sure a `partial_fit()` method makes sense for Word2Vec - incremental training is at best experimental, and not at all the usual mode of operation.